File: readability/readability.c
#include <cs50.h>
#include <stdio.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int _isalpha(int c);


int main(void)
{
    int letters, words, sentences;
    float L = 0, S = 0, index;

    string text = get_string("Text: ");
    letters = count_letters(text);
    if (letters != 0)
    {
        words = count_words(text) + 1;
    }
    else
    {
        words = count_words(text);
    }
    sentences = count_sentences(text);



    if (words != 0)
    {
        L = letters / (float)words * 100;
        S = sentences / (float)words * 100;
        index = 0.0588 * L - 0.296 * S - 15.8;
    }
    else
    {
        index = 0;
    }

    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)(index + 0.5));
    }


}

int count_letters(string text)
{
    int i = 0;
    int s = 0;

    while (text[i] != '\0')
    {
        if (_isalpha(text[i]))
        {
            s++;
        }
        i++;
    }
    return (s);
}
int count_words(string text)
{
    int i = 0;
    int s = 0;

    while (text[i] != '\0')
    {
        if (text[i] == ' ')
        {
            s++;
        }
        i++;
    }
    return (s);
}
int count_sentences(string text)
{
    int i = 0;
    int s = 0;

    while (text[i] != '\0')
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            s++;
        }
        i++;
    }
    return (s);
}

int _isalpha(int c)
{
    if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z'))
    {
        return (1);
    }
    return (0);
}

File: sentimental-readability/readability.py
# get the input text from the usr
text = input("Text: ")

letters = 0
words = 0
sentences = 0

for l in text:
    if (ord(l) >= ord('A') and ord(l) <= ord('Z')) or (ord(l) >= ord('a') and ord(l) <= ord('z')):
        letters = letters + 1
    if l == ' ':
        words = words + 1
    if l == '.' or l == '!' or l == '?':
        sentences = sentences + 1

if words != 0:
    words = words + 1

# the Coleman-Liau formula
L = (letters / words) * 100
S = (sentences / words) * 100
x = 0.0588 * L - 0.296 * S - 15.8
# print("letters {}\nwords {}\nsentences {}".format(letters, words, sentences))

if x < 1:
    print('Before Grade 1')
elif x >= 16:
    print('Grade 16+')
else:
    print("Grade {}".format(int(x + 0.5)))

